device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/CorrelatingMessageCallback.java

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.device;
+
+import com.microsoft.azure.sdk.iot.device.transport.IotHubTransportPacket;
+
+/**
+ * This interface has a number of call backs to notify when a message is in a specific part of the {@code IotHubTransport} lifecycle.
+ */
+public interface CorrelatingMessageCallback
+{
+
+
+    IotHubClientProtocol protocol = null;
```

Comment:
Is this used anywhere? I wouldn't expect that this class would need to know the protocol in use since both AMQP and MQTT behave similarly when it comes to sending a message, getting an ack, and finally getting a status code message

Reply:
Correct, before I reached out about the request id being an integer I was using this switch between correlationid and requestid